hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -200,6 +202,7 @@ public Node(HazelcastInstanceImpl hazelcastInstance, Config config, NodeContext
             securityContext = config.getSecurityConfig().isEnabled() ? nodeExtension.getSecurityContext() : null;
 
             nodeEngine = new NodeEngineImpl(this);
+            config.setConfigurationService(nodeEngine.getConfigurationService());
```

Comment:
`DynamicConfigurationAwareConfig.configurationService` field is not `volatile`? How is this write guaranteed to be observed by readers? `final` field assignment is done earlier `this.config = config`.